config/foreman-hiera.yaml

```diff
@@ -8,6 +8,8 @@ hierarchy:
     path: "custom.yaml"
   - name: "Kafo Answers"
     path: "%{facts.kafo.scenario.answer_file}"
+  - name: "Scenario defaults"
+    path: "scenario/%{facts.kafo.scenario.id}.yaml"
```

Comment:
One downside to this approach is that the scenario must provide this file here or add it to the installed location for this repository. If the scenario is stored in it's own repository and delivered in its own RPM that is a bit uglier.

What are your thoughts on letting this be driven through the scenario's configuration file via  variable that denotes what file to look in or through a convention that introduces a new file e.g. `katello-defaults.yaml`

Reply:
Thinking out loud now.

Rather than another file, we could also store it in the scenario itself. That lets the package deal with it. The downside is that you now need migrations where with a standalone file you can just drop the file in place.

Another is using the custom fact storage to point to some other file. I now realize I didn't document this in Kafo's README.md, but https://github.com/theforeman/kafo/commit/414bc6d2c21ecf1424c08589848a3f1668377da3 should be straight forward enough. Then in foreman-hiera.yaml you can add a layer `%{facts.kafo.scenario.custom.overrides}` or similar.

However, this indirection does add a lot of complexity where dropping the single file in `/usr/share/foreman-installer/config/scenario/my_own_scenario.yaml` is much more straight forward. That means it's easier to follow and debug which makes it still my preference.

On a side note: longer term I'd prefer for all scenarios to be installed to `/usr/share/foreman-installer/scenarios.d`, including addons. For that I'd like to write a small helper script that takes a scenario file, an answers file, migrations directory and defaults file to place it in the right layout. This helper should be provided by foreman-installer.